src/credentials/FabricTable.h

```diff
@@ -202,7 +205,24 @@ class DLL_EXPORT FabricInfo
         mFabricIndex = kUndefinedFabricIndex;
     }
 
-    CHIP_ERROR SetFabricInfo(FabricInfo & fabric);
+    /**
+     * Verify the validity of the passed fabric info, and then emplace into
+     * this.  If a policy is passed, enact this for the fabric info validation.
+     *
+     * @param fabric fabric to emplace into this
+     * @param policy validation policy to apply, or nulllptr for none
+     * @return CHIP_NO_ERROR on success, else an appopriate CHIP_ERROR
+     */
+    CHIP_ERROR SetFabricInfo(FabricInfo & fabric, Credentials::CertificateValidityPolicy * policy);
+
+    /**
+     * Verify the validity of the passed fabric info, and then emplace into
+     * this.
+     *
+     * @param fabric fabric to emplace into this
+     * @return CHIP_NO_ERROR on success, else an appopriate CHIP_ERROR
+     */
+    CHIP_ERROR SetFabricInfo(FabricInfo & fabric) { return SetFabricInfo(fabric, nullptr); }
```

Comment:
So a question.  Why would we not apply our "gather the minimum notbefore" policy in this case?

Reply:
This is called from the controller, which is installing its own certificates.  The particular path in question doesn't update Last Known Good Time at all.

The way the controller commissions itself is extremely convoluted and not something I'm willing to remediate.  It shouldn't be calling SetFabricInfo in the first place.

But for the purposes here, I would just say that 1) if the controller is doing an end-run around the FabricTable and updating FabricInfos directly, it's not really possible for the FabricTable to arbitrate a situation like this where visibility of all fabrics is required (Last Known Good Time is such a situation).

But also 2) the requirement to update Last Known Good Time at the time of commissioning is referring to in-band Matter commissioning, and the controller is doing a self-commissioning, which the spec can't really speak to.

I don't consider this a bug introduced by this PR, but rather it highlights an abusive thing the controller code is already doing.

Reply:
Ah, I see now that https://github.com/project-chip/connectedhomeip/pull/18866/files is also using the SetFabricInfo interface.

I will rebase on #18866 and change this so the update happens with a `FabricTable::UpdateFabricInfo(FabricIndex fabricIndex, FabricInfo fabricInfo)` call or similar so that FabricTable can do what's needed.

Reply:
Done.